.pre-commit-config.yaml

```diff
@@ -1,11 +1,40 @@
 exclude: '^$'
 fail_fast: false
 repos:
--   repo: https://github.com/pycqa/flake8
-    rev: 6.1.0
-    hooks:
-    - id: flake8
-      additional_dependencies: [flake8-docstrings, flake8-debugger, flake8-bugbear]
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: 'v0.1.6'
+  hooks:
+  - id: ruff
+# once https://github.com/astral-sh/ruff/issues/2402 is fully resolved then we can get rid of flake8:
+- repo: https://github.com/pycqa/flake8
+  rev: 6.1.0
+  hooks:
+  - id: flake8
+    additional_dependencies: [flake8-docstrings, flake8-debugger, flake8-bugbear, mccabe]
```

Comment:
I think mccabe and maybe bugbear at least are fully covered by ruff, so they could be remove from flake8

Reply:
Ok. I can remove that.